dashboard/modules/job/sdk.py

```diff
@@ -296,6 +297,8 @@ def submit_job(
         metadata.update(self._default_metadata)
 
         self._upload_working_dir_if_needed(runtime_env)
+        # Run the RuntimeEnv constructor to parse local pip/conda requirements files.
+        runtime_env = RuntimeEnv(**runtime_env).to_dict()
```

Comment:
Do this constructor also do working_dir and py_modules uploading? For example, if this and the previous line were reversed what would happen?

Ideally we would unify the working_dir uploading part so we don't have these divergent codepaths. For example, we could pass in an optional `upload_working_dir` callback into the constructor, which in this case would use HTTP.

Reply:
The constructor doesn't do any uploading, it always assumes everything is a URI already in the input.  So if you switched the lines, it would fail.  I think this status quo is okay, but I agree we ideally wouldn't have to always call upload_py_modules and upload_working_dir before calling the runtime env constructor.  Maybe we can refactor in the future